Tiltfile

```
@@ -354,6 +354,10 @@ def deploy_observability():
         k8s_yaml(read_file("./.tiltbuild/yaml/prometheus.observability.yaml"), allow_duplicates = True)
         k8s_resource(workload = "prometheus-server", new_name = "prometheus", port_forwards = "9090", extra_pod_selectors = [{"app": "prometheus"}], labels = ["observability"])
 
+    if "capi-state-metrics" in settings.get("deploy_observability", []):
+        k8s_yaml(read_file("./.tiltbuild/yaml/capi-state-metrics.observability.yaml"), allow_duplicates = True)
+        k8s_resource(workload = "capi-state-metrics", new_name = "capi-state-metrics", port_forwards = "8081:8080", extra_pod_selectors = [{"app": "kube-state-metrics"}], labels = ["observability"])
```

```suggestion
        k8s_resource(workload = "capi-state-metrics", new_name = "capi-state-metrics", port_forwards = "8081:8080", extra_pod_selectors = [{"app": "kube-state-metrics"}], labels = ["observability"])
```

TODO: decide on which port to use, I think we should not block 8080?!

---

We don't need to expose this on localhost though, right? We can just make it available through a service in the cluster instead of needing a port-forward, so the port picked shouldn't really matter. Am I missing something?

---

Good point :-) It is already available inside the cluster and gets scraped by prometheus 👍